### PR TITLE
The great refactor

### PR DESCRIPTION
**Pull request by SimoneAriens**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,20 +63,56 @@ experiment:
...
-      n_splits: 2
-    glass:
-      n_splits: 2
   preprocessor:
     - 'dummy'
     - 'rank_transformer'
 ```
-When executing `python run.py` an experiment for all possible combination of parameters will be executed. The results for each experiment (metrics + plots)
-will be stored in separate folders within the `output` folder.
+When executing `python run.py -d glass` an experiment for all possible combination of parameters will be executed on the glass dataset.
```
Comment (rolfypma-nfi) [inline README.md:70]: nieuw kaartje om meerdere datasets mogelijk te maken?

File: lrbenchmark/data/models.py [hunk 2]
```diff
@@ -52,30 +48,23 @@ class MeasurementPair:
     """
     measurement_a: Measurement
     measurement_b: Measurement
-    extra: Mapping[str, Any]
+    extra: Mapping[str, Any] = None
+    value: Optional[np.ndarray] = None
```
Comment (rolfypma-nfi) [inline lrbenchmark/data/models.py:52]: is value something that will exist alongside 'score's? Or is it a 2xn matrix that concatenates the vectors of the measurements? I think we need an explanation here of what is intended with the value (or scores)

Comment (rolfypma-nfi) [reply lrbenchmark/data/models.py:52]: or do we get rid of 'score' (my preference)?

File: lrbenchmark/pairing.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    def transform(self, measurements: Iterable[Measurement], seed: Optional[int] = None) -> List[MeasurementPair]:
+        raise NotImplementedError
+
+
+class CartesianPairing(BasePairing):
+    """
+    Creates pairs of instances.
+
+    This transformer takes a list of Measurement as input, and returns a list of MeasurementPair.
+    By default, the list of MeasurementPair contains all possible pairs of Measurement.
```
Comment (rolfypma-nfi) [inline lrbenchmark/pairing.py:26]: does this include the measurement with itself?

File: run.py [hunk 4]
```diff
@@ -11,76 +11,79 @@
...
+    lr_metrics = calculate_lr_statistics(*Xy_to_Xn(validate_lrs, validate_labels))
 
-    if not isinstance(calibrated_scorer.scorer, DummyClassifier):
-        test_probas = np.concatenate(test_probas)
-        test_predictions = np.concatenate(test_predictions)
-        results['auc'] = roc_auc_score(test_labels, test_probas)
-        results['acc'] = accuracy_score(test_labels, test_predictions)
+    results = {'desc': get_experiment_description(selected_params), 'figures': figs, **lr_metrics._asdict()}
+    # todo fix scores -> between 0 and 1
+    if np.min(validate_probas) >= 0 and np.max(validate_probas) <= 1:
```
Comment (rolfypma-nfi) [inline run.py:80]: roc works with values outside [0,1], suggest to remove the todo and if statement?

File: run.py [hunk 5]
```diff
@@ -11,76 +11,79 @@
...
     """
-    calibrated_scorer = lir.CalibratedScorer(scorer, calibrator)
-
-    test_lrs = []
-    test_labels = []
-    test_probas = []
-    test_predictions = []
+    validate_lrs = []
+    validate_labels = []
+    validate_probas = []
```
Comment (rolfypma-nfi) [inline run.py:41]: these are currently not necessarily in [0,1]. Better to call them scores?

File: lrbenchmark/data/models.py [hunk 6]
```diff
@@ -52,30 +48,23 @@ class MeasurementPair:
...
+    def measurements(self) -> Tuple[Measurement, Measurement]:
+        return self.measurement_a, self.measurement_b
 
     def get_x(self) -> np.ndarray:
-        if 'score' in self.extra.keys():
-            return np.array([self.score]) if not isinstance(self.score, np.ndarray) else self.score
-        raise ValueError("No score found in the extra mapping.")
+        return np.vstack([self.measurement_a.get_x(), self.measurement_b.get_x()]).T
 
     def get_measurement_values(self) -> np.ndarray:
```
Comment (cdwagenaar) [inline lrbenchmark/data/models.py]: Overlegd en weggehaald

**Review by cdwagenaar — APPROVED** [review]

**Pull request merged**